pkg/cnab/config-adapter/stamp.go

```diff
@@ -75,9 +78,11 @@ func (c *ManifestConverter) GenerateStamp() (Stamp, error) {
 	stamp.EncodedManifest = base64.StdEncoding.EncodeToString(rawManifest)
 
 	// Remember the mixins used in the bundle
-	stamp.Mixins = make(map[string]MixinRecord, len(c.Manifest.Mixins))
-	for _, m := range c.Manifest.Mixins {
-		stamp.Mixins[m.Name] = MixinRecord{}
+	stamp.Mixins = make(map[string]MixinRecord, len(c.Mixins))
```

Comment:
Here you are looping over the list of installed mixins, not the list of mixins used in the bundle. You need to loop over `c.Manifest.Mixins` which are the mixins used by the bundle. Then match the mixin name against the list of installed mixins `c.Mixins` to lookup the version number.

Feel free to rename c.Mixins to c.InstalledMixins to help clear up confusion between those two variables.

The other problem is that the existing unit tests that test this functionality are not passing in a list of installed mixins. You'll need to change the tests like so:

```go
     installedMixins := []mixin.Metadata{
		{Name: "exec", VersionInfo: pkgmgmt.VersionInfo{Version: "v1.2.3"}},
	}
	a := NewManifestConverter(c.Context, m, nil, installedMixins)
```

Reply:
Once you are done, you either need to update the test files with `mage updateTestFiles` or edit pkg/cnab/config-adapter/testdata/mybuns.bundle.json by hand so that it includes the mixin version number.